src/liblzma/check/crc64_fast.c

```diff
@@ -438,35 +438,47 @@ is_clmul_supported(void)
 }
 
 
-#ifdef HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR
-#	define CRC64_FUNC_INIT
-#	define CRC64_SET_FUNC_ATTR __attribute__((__constructor__))
-#else
-#	define CRC64_FUNC_INIT = &crc64_dispatch
-#	define CRC64_SET_FUNC_ATTR
-static uint64_t crc64_dispatch(const uint8_t *buf, size_t size, uint64_t crc);
-#endif
+#ifdef HAVE_FUNC_ATTRIBUTE_IFUNC
+#	ifdef __cplusplus
+// The extern variable "C" is required to work with C++'s function name 
+// mangling in C++. If this is not included, the "crc64_resolver" will be
+// mangled and won't compile.
+extern "C" {
+#	endif
 
+// The resolver must accept no arguments (void), and must specify the return and
```

Comment:
We try to keep an 80 character width in this project. Can you wrap these comment lines at 80?